### ISSUE TITLE
Help with making it work on Windows xp 32bit

### ISSUE THREAD

**Issue by takicreepy** [olauncher/olauncher#82]
Info: Open JDK 21.0.3+12-LTS 32bit
![imagem_2024-06-03_213809099](https://github.com/olauncher/olauncher/assets/157829455/319bfc87-3482-47fb-9ff5-4224f03ba7bf)

**Comment by bigfoot547** [comment]
well that specific error is because it couldn't look up `launcher.mojang.com`. Is this computer connected to the internet (hopefully not thinking about it)? if not, the redistributable can't download the original launcher jar to patch it.

**Comment by takicreepy** [comment]
1. yes this computer is totally connected to internet, which can sound strange but isnt a real problem, i know what i download and my modem has a great firewall, so i got no problems til this day
2. on onecoreapi github page said that you could install liberica jre without needing nothing, just install it and it works, i tried and it... worked- at least as far as i know (https://bell-sw.com/pages/downloads/#jdk-21-lts if you want the link)

**Comment by bigfoot547** [comment]
hmm the fact that it couldn't look up the domain is strange then. What happens when you visit https://launcher.mojang.com/ in a browser? (it should show an XML error page)

if your browser can't look up the domain, it might be your system's DNS setup. If it can, maybe it's a java issue.

**Comment by takicreepy** [comment]
My computer is turned off right now, but i tried doing this earlier and it showed the xml page

**Comment by bigfoot547** [comment]
okay, was the launcher error a one-time thing or does it happen every time you run the launcher? if there's a temporary outage while the launcher is opening, there will be issues like that.

Debugging this is hard because the issue is quite possibly upstream of any olauncher code

**Comment by takicreepy** [comment]
every time, but yea its very hard, i thought of just copying the files that olauncher creates on .minecraft, but my other pc is x64 so i think the created data is different? i dont know even if this would work, if you want me to perform some test i can do it here

**Comment by bigfoot547** [comment]
Nope, java bytecode is java bytecode. It doesn't matter what platform it is compiled on. It should work if you copy over the autool directory.

**Comment by takicreepy** [comment]
![image](https://github.com/olauncher/olauncher/assets/157829455/3c4bab5e-ec6a-46c0-839b-b3d9af3e9bc5)
i copied the autool directory, and now it opens, but now i got this error on the launcher when i try to login with the options

**Comment by bigfoot547** [comment]
alright, that is because your JVM's truststore is out of date

**Comment by takicreepy** [comment]
3 things i discovered
1. the java was opening on java 8u45 _(as you see i didnt opened it from console this time so it didnt used java 21)_ that came with the toshiba recovery cd when i formatted this notebook last time

2. i get strange errors with different versions of java, check it (when i click ok on the microsoft failed authentication it does the unhandled thing on **java 17 and 11** i think its hard to know cause my cursor didnt appeared for some reason), (**i cutted some parts of the video** cause this computer is not so fast when recording so i had to cut some things and make others straight to the point, but **i didnt cutted any part of the log or something like that, i just cutted the loadings that were mostly freezed screens**)

3. java detects my pc as windows 7, which i dont know if its cause i installed onecoreapi? i not so sure cause im not using compatibility mode on java.exe neither i used when i installed it (its impossible cause its a .msi installer not exe)

over all:
java 21 crashes when click login
java 17 & 11 fails when authenticating

video: https://www.mediafire.com/file/j0j4eparrl7n0n4/Meu_Filme%257E1.mp4/file

**Comment by bigfoot547** [comment]
I see a lot of "Connection timed out" so maybe there is an issue there. Also, use the device code option for logging in. It is way more likely to work.

**Comment by notbronwyn** [comment]
Try java 8, I’ve had connectivity errors with java 9+ on XP and Vista

**Comment by bigfoot547** [comment]
here's a forum post that details the process of getting this working https://board.eclipse.cx/viewtopic.php?t=707
